synapse/handlers/pagination.py

```diff
@@ -527,7 +527,6 @@ async def get_messages(
             return {
                 "chunk": [],
                 "start": await from_token.to_string(self.store),
-                "end": await next_token.to_string(self.store),
```

Comment:
I think for this to work correctly, we need to move this block further up (above the call to `event_filter.filter`.

The problem is that we hit this block if there are no *visible* events between the two tokens, but that might just be because the `event_filter` has filtered them all out. If we paginate further, we might find more events.

Reply:
Hmmm, that looks correct, I'll push another commit to fix that